vrs/marten-construction-script.py

```diff
@@ -40,191 +57,124 @@
 x_freq = ht_whole.aggregate(hl.agg.count_where(ht_whole.locus.contig == "chrX")) / whole_count
 y_freq = ht_whole.aggregate(hl.agg.count_where(ht_whole.locus.contig == "chrY")) / whole_count
 
-print(f'Frequencies as: indels {indel_freq} and long ref {ref_freq} and long var {var_freq} and x freq {x_freq} and y freq {y_freq}')
+logger.info("Indel Frequency: %f , Ref Frequency: %f , Var Frequency: %f , X Frequency: %f , Y Frequency: %f", 
+            indel_freq, ref_freq, var_freq, x_freq, y_freq)
 
-Quite time-intensive, so ran previously to save computation headache
+print(f'Frequencies as: indels {indel_freq} and long ref {ref_freq} and long var {var_freq} and x freq {x_freq} and y freq {y_freq}')
 
-Feel free to run yourself - at your own peril ! 
 """
-
 indel_freq = 0.14525709561723196
 ref_freq = 0.02122722096390106
-var_freq = 0.0381956465303165
+var_freq = 0.0381956465303165 
 x_freq = 0.0398928560027597
 y_freq = 0.0015374206699161612
 whole_count = 759302267
-
-# e.g.: for 1000 variants, 145 will be indels
-# for 1000 variants, 40 will be X chromosomes and only 1.5 will be Y chromosome
-
-# Set a temporary path to make writing checkpoints easier (and more easy to edit for future users)
-tmp_path = "gs://gnomad-tmp-4day/marten-temp-01-18-v3/"
-
-# Construct tables of samples from this
-# Makeup: 50k random, 10k extra indels, 10k more long references, 10k long variants, 10k sex chromosome
-# --> (9k X, 1k Y)
-
-# 50,000 random variants
+# e.g.: For 1000 variants, 145 will be indels
+# for 1000 variants, 40 will be on chromosome X and only 1.5 will be on chromosome Y
+
+# Construct Tables for each of the targeted variant types desired for the test subset:
+# 50k random, 10k extra indels, 10k additional long references, 10k  additional long variants, 10k on sex chromosomes (9k on X, 1k on Y)
+
+# Create freq dictionary with variant type as the key and pre-calculated frequencies as the value
+freq_dict = {"indel": indel_freq,
+    "long_ref": ref_freq,
+    "long_var": var_freq,
+    "on_x":  x_freq,
+    "on_y": y_freq}
+
+# Create desired counts dictionary with variant type as the key and desired number of that variant type in the test subset as the value
+desired_counts = {"indel": 10000,
+    "long_ref": 10000,
+    "long_var": 10000,
+    "on_x":  9000,
+    "on_y": 1000}
+
+# Note: It is important to define everything after this (like the random samples) since if you dont, fields dont match
```

Comment:
can you clarify what this means: "define everything after this (like the random samples)"

Reply:
Oh sorry that was unclear, previously I had a problem where I was sub-sampling the random samples BEFORE I did the annotations for the other populations, and that led to problems when those tables had the four additional annotations but the random sample table didn't. The comment was me trying to say that I needed to define all of my random samples AFTER the block of code that this preceded (the annotations) , though I worded it a bit unclearly